codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenUtils.java

```diff
@@ -130,4 +131,40 @@ public static boolean isJsonMediaType(String mediaType) {
     public static String getSyntheticTypeNamespace() {
         return CodegenUtils.SYNTHETIC_NAMESPACE;
     }
+
+    /**
+     * Get the pointer reference to operand , if symbol is pointable.
+     * This method can be used by deserializers to get pointer to
+     * operand.
+     *
+     * @param symbol The Symbol corresponding to shape whose value needs to be assigned.
+     * @param operand The Operand is the value to be assigned to the symbol shape.
+     * @return The Operand, along with pointer reference if applicable
+     */
+    public static String generatePointerReferenceIfPointable(Symbol symbol, String operand) {
```

Comment:
It  would be better to just make this another formatter that works like the P formatter. D for dereference?

Reply:
Good idea, will do that.

Reply:
That would require making symbols for variables, as the symbol for the shape itself is not what you take a reference to. It's a named instance of a particular symbol that you are taking a reference to. Not sure if it this makes sense to encapsulate as a formatter, unless we create symbols for operand variables, and add a property that states that it is a variable not a shape symbol.

Reply:
So looks like Sean@ has a PR with similar changes, that can be used here. Instead of doubling the work and making rebasing difficult, I can add a TODO here to use those changes directly once merged.

Reply:
I have updated the PR to use `isShapePassByReference` to correctly consider other shapes. Having a formatter for variable might be an overkill as would need to make it a symbol, but I will try to prototype and see if we can replace this with a formatter.